### PR TITLE
[coverage] More unit tests for scapy/*.py files

### PR DESCRIPTION
**Pull request by guedou**
This PR add systematic unit tests to `scapy/*.py` files. It also fixes one issue and removes a useless function.

### CODE REVIEW

**Comment by gpotter2** [comment]
If you want to fix the Route6 bugs on windows, you may consider first merging https://github.com/secdev/scapy/pull/398 before performing the fixes, as they surely will conflicts each other...

**Comment by gpotter2** [comment]
I would recommand you to try running your tests using something like:
```
def get_fake_adapter():
    if WINDOWS:
        data = {}
        data['name'] = "dummy0"
        data['description'] = "Does not exist"
        data['win_index'] = -1
        data['guid'] = "{0XX00000-X000-0X0X-X00X-00XXXX000XXX}"
        data['invalid'] = True
        return NetworkInterface(data)
    else:
        return 'dummy0'
```

**Comment by guedou** [comment]
@gpotter2 thanks. It fixes some problems, but triggers other ones =) 
I need to spend some time fixing bugs.

**Comment by guedou** [comment]
@p-l- this PR is finished and only asks to be reviewed.

File: scapy/arch/bpf/core.py [hunk 1]
```diff
@@ -53,12 +53,14 @@ def get_if_raw_addr(ifname):
     try:
         fd = os.popen("%s %s" % (conf.prog.ifconfig, ifname))
     except OSError, msg:
-        raise Scapy_Exception("Failed to execute ifconfig: (%s)" % msg)
+        warning("Failed to execute ifconfig: (%s)" % msg)
+        return "\0\0\0\0"
 
     # Get IPv4 addresses
     addresses = [l for l in fd.readlines() if l.find("netmask") >= 0]
```
Comment (p-l-) [inline scapy/arch/bpf/core.py]: can you use this opportunity to remove `.readlines()` useless list creation? (`addresses = [l for l in fd if l.find("netmask") >= 0]`)

File: scapy/route6.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def delt(self, dst, gw=None):
         l = filter(lambda x: in6_ptop(x[0]) == dst and x[1] == plen, self.routes)
         if gw:
             gw = in6_ptop(gw)
-            l = filter(lambda x: in6_ptop(x[0]) == gw, self.routes)
+            l = filter(lambda x: in6_ptop(x[2]) == gw, self.routes)
```
Comment (p-l-) [inline scapy/route6.py]: Maybe you could change this to `l = [x for x in self.routes if in6_ptop(x[2]) == gw`?

File: scapy/utils6.py [hunk 3]
```diff
@@ -395,7 +359,7 @@ def in6_getLocalUniquePrefix():
     tod = struct.pack("!II", i,j)
     # TODO: Add some check regarding system address gathering
     from scapy.arch import get_if_raw_hwaddr
-    rawmac = get_if_raw_hwaddr(conf.iface6)[1]
+    rawmac = get_if_raw_hwaddr(conf.iface)[1]
```
Comment (p-l-) [inline scapy/utils6.py]: Why do you change this?

File: scapy/arch/bpf/core.py (hunk 1 above)
Comment (guedou) [inline scapy/arch/bpf/core.py]: Done.

File: scapy/route6.py (hunk 2 above)
Comment (guedou) [inline scapy/route6.py]: Done.

File: scapy/utils6.py (hunk 3 above)
Comment (guedou) [inline scapy/utils6.py]: It allows the function to work without a working `conf.iface6`. It is silly anyway. I used `RandMac` instead as the RFC does not specify which interface must be used.

**Comment by gpotter2** [comment]
@guedou You should have a look at the branch hisory 😄

**Comment by guedou** [comment]
@p-l- the PR is OK now.

**Pull request merged**